src/Entities/Misc/RailedMoveBlock.cs

```diff
@@ -308,22 +323,38 @@ public override void Update()
 
         if (dir != Vector2.Zero)
         {
+
             float newSpeed = 0f;
 
             icon = idleIcon;
-            if ((topPressed || bottomPressed) && HasPlayerRider() && Input.MoveY.Value != 0)
+            if (easedVerticalMovement)
             {
-                newSpeed = moveSpeed * Input.MoveY.Value * (dir.Y > 0f ? 1f : -1f);
-                newFillColor = MoveBgFill;
-                icon = Input.MoveY.Value == 1 ? DownIcon : UpIcon;
+                if ((topPressed || bottomPressed) && HasPlayerRider() && (Input.Feather.Value.Y > 0.4f || Input.Feather.Value.Y < -0.4f))
+                {
+                    newSpeed = moveSpeed * Math.Sign(Input.Feather.Value.Y) * (dir.Y > 0f ? 1f : -1f);
+                    newFillColor = MoveBgFill;
+                    icon = Input.MoveY.Value == 1 ? DownIcon : UpIcon;
```

Comment:
```suggestion
                    icon = Math.Sign(Input.Feather.Value.Y) == 1 ? DownIcon : UpIcon;
```

We should use the sign of `Input.Feather.Value.Y` to determine the direction the player is moving in, because we could go down using using the value from `Input.Feather.Value.Y` while `Input.MoveY.Value`  is still zero.

Reply:
fixed and tested